src/output/plugins/httpd/HttpdOutputPlugin.cxx

```diff
@@ -44,6 +44,8 @@ HttpdOutput::HttpdOutput(EventLoop &_loop, const ConfigBlock &block)
 			ServerSocket::SetDscpClass(value);
 		});
 
+	password = block.GetBlockValue("password", "");
```

Comment:
Why not `nullptr` for "no password necessary"? That seems like a natural choice, and would make your code simpler.

Reply:
> Why not `nullptr` for "no password necessary"? That seems like a natural choice, and would make your code simpler.

Because MPD segfaults at startup if test for nulllptr and mpd.conf lacks requested parameter.

src/playlist/plugins/SoundCloudPlaylistPlugin.cxx (updated to support new SoundCloud API. PR soon)

```
(...)
static std::string token;
(...)

static bool
soundcloud_init(const ConfigBlock &block)
{
        token = block.GetBlockValue("auth_token", nullptr);
        if (token.empty()) {
(...)
```


mpd.conf
```
playlist_plugin {
        name "soundcloud"
        enabled"yes"
        oauth_token "secret"
}
```

Testing for "" results in config: option 'oauth_token' on line X was not recognized) logentry and MPD continues normal operation.

Reply:
> Because MPD segfaults at startup if test for nulllptr and mpd.conf lacks requested parameter.

.... what?? I don't get it. Why does MPD crash? Because MPD is buggy or because this new code is buggy? Or because the code you imagine is buggy?

Reply:
@MaxKellermann maybe I misunderstood you. Your suggestion was to pass block.GetBlockValue("something", nullptr) as the second argument nullptr. In the event that "something" does not exist in mpd.conf, a segfault occurs at startup. I can't say whether it's an MPD bug.

Reply:
If you just replace `""` with `nullptr`, of course it will crash. That's not how code hacking works, if you just replace something without thinking and without understanding the code.

Reply:
Yeah, I agree it is nicer. Implemented.